components/AdminPanel/AdminPanel.tsx

```diff
@@ -22,20 +23,22 @@ const columns = [
 ] as const;
 
 export const AdminPanel = () => {
-  const router = useRouter();
-  const isPublic = (router.query.isPublic || 'all') as IsPublic;
+  const {
+    query: { isPublic },
+    changeQuery,
+  } = useSmartQuery(
+    object({ isPublic: string().matches(/(true|false)/, { excludeEmptyString: true }) }),
+  );
 
-  const { value: blogs } = useQuery(useCallback(() => getBlogs(isPublic), [isPublic]));
+  const { value: blogs } = useQuery(useCallback(() => getBlogs(isPublic as IsPublic), [isPublic]));
```

Comment:
Po co rzutowanie?

Reply:
Typ `isPublic` to `string | undefined`
Jak bez rzutowania przypisać go do `"true"|"false"|undefined`?